rpm/0002-Add-D-Bus-interface-for-querying-client-process-deta.patch

```diff
@@ -0,0 +1,207 @@
+the client behind the proxy connection.
+
+Implement org.sailfishos.sailjailed.Identify() method call that returns
+pid, uid, and other details about the connected client.
+
+D-Bus policy configuration allowing/denying such method calls to be made
+is assumed to be defined elsewhere.
+
+Signed-off-by: Simo Piiroinen <simo.piiroinen@jolla.com>
+---
+ flatpak-proxy.c | 138 +++++++++++++++++++++++++++++++++++++++++++++++-
+ 1 file changed, 136 insertions(+), 2 deletions(-)
+
+diff --git a/flatpak-proxy.c b/flatpak-proxy.c
+index 8bf9a4f..4ca3223 100644
+--- a/flatpak-proxy.c
++++ b/flatpak-proxy.c
+@@ -22,6 +22,7 @@
+ 
+ #include <unistd.h>
+ #include <string.h>
++#include <stdio.h>
+ 
+ #include "flatpak-proxy.h"
+ 
+@@ -1998,7 +1999,7 @@ update_socket_messages (ProxySide *side, Buffer *buffer, Header *header)
+ }
+ 
+ static void
+-queue_fake_message (FlatpakProxyClient *client, GDBusMessage *message, ExpectedReplyType reply_type)
++queue_bus_message (FlatpakProxyClient *client, GDBusMessage *message)
+ {
+   Buffer *buffer;
+ 
+@@ -2009,6 +2010,12 @@ queue_fake_message (FlatpakProxyClient *client, GDBusMessage *message, ExpectedR
+   g_object_unref (message);
+ 
+   queue_outgoing_buffer (&client->bus_side, buffer);
++}
++
++static void
++queue_fake_message (FlatpakProxyClient *client, GDBusMessage *message, ExpectedReplyType reply_type)
++{
++  queue_bus_message(client, message);
+   queue_expected_reply (&client->client_side, client->last_serial, reply_type);
+ }
+ 
+@@ -2306,6 +2313,129 @@ handle_deny:
+     queue_initial_name_ops (client);
+ }
+ 
++static gboolean
++handle_proxy_method(FlatpakProxyClient *client, Header *header, Buffer *buffer)
++{
++  if (header->type != G_DBUS_MESSAGE_TYPE_METHOD_CALL)
++    return FALSE;
++
++  if (g_strcmp0 (header->interface, "org.sailfishos.sailjailed"))
++    return FALSE;
++
++  GDBusMessage *message = g_dbus_message_new_from_blob (buffer->data, buffer->size,
++                                                        G_DBUS_CAPABILITY_FLAGS_NONE, NULL);
++  if (message)
++    {
++      GDBusMessage *reply = NULL;
++      if (!g_strcmp0(header->member, "Identify"))
++        {
++          GVariantBuilder *array_builder = g_variant_builder_new(G_VARIANT_TYPE_ARRAY);
++          if (client->client_side.connection)
++            {
++              GSocket *sck = g_socket_connection_get_socket(client->client_side.connection);
++              if (sck)
++                {
++                  GCredentials *creds = g_socket_get_credentials(sck, NULL);
++                  if (creds)
++                    {
++                      gint32 uid = g_credentials_get_unix_user(creds, NULL);
++                      if (uid != -1)
++                        g_variant_builder_add(array_builder, "{sv}", "uid", g_variant_new_int32(uid));
++                      gint32 pid = g_credentials_get_unix_pid(creds, NULL);
++                      if (pid != -1)
++                        {
++                          // pid
```

Comment:
Wouldn't it better to split it to the separate functions to retrieve information?:
* // proc/pid/exe
* // proc/pid/comm
* // proc/pid/cmdline
* // proc/pid/maps

Reply:
Depends. Originally the idea was to have the patch as self contained as possible. But now that more invasive changes were needed anyway -> used helper functions like you suggested.